### PR TITLE
Create Lao by attendee should fail

### PR DESCRIPTION
**Pull request by Tesa27**
#883 is the base of this PR everything in there is in here, but some tests were not passing for some reason.

- Additionally in this PR we are construction the ids using a hash function instead of feeding raw predetermined ids, this way we can check if the backend checks well the computation of the ids as well

### CODE REVIEW

**Comment by Tesa27** [comment]
#972 will address the failing be2 tests

File: tests/karate/src/test/java/be/LAO/create.feature [hunk 1]
```diff
@@ -100,3 +100,22 @@ Feature: Create a pop LAO
...
     Then match answer contains INVALID_MESSAGE_FIELD
     And match frontend.receiveNoMoreResponses() == true
+    
+  Scenario: Create Lao with different public key from the organizer should fail with error response
+    Given def laoCreateRequest =
+      """
+        {
+          "object": "lao",
+          "action": "create",
+          "id": '#(getLaoValid)',
```
Comment (pierluca) [inline tests/karate/src/test/java/be/LAO/create.feature:110]: how's `getLaoValid` computed ? Does it include the `getOrganizer` sender ?

File: be1-go/channel/lao/verification.go [hunk 2]
```diff
@@ -106,7 +106,7 @@ func (c *Channel) verifyMessageRollCallCreate(rollCallCreate *messagedata.RollCa
 		rollCallCreate.Name,
 	)
 	if rollCallCreate.ID != expectedID {
-		return xerrors.Errorf("roll call id is %s, should be %s", rollCallCreate.ID, expectedID)
+		return xerrors.Errorf("%s|roll call id is %s, should be %s", strings.ReplaceAll(c.channelID, messagedata.RootPrefix, ""), rollCallCreate.ID, expectedID)
 	}
```
Comment (GabrielFleischer) [inline be1-go/channel/lao/verification.go:110]: Why is this piece of code here ?

File: tests/karate/src/test/java/be/constants.feature [hunk 3]
```diff
@@ -2,34 +2,55 @@
...
       """
-    * def createRollCallInvalid =
-      """
-        function(){
-          return "Dui7C1LBEXlRC8ItV2B0zWfUSD6YiGJt6N_I_m02uw4="
+
+    * def getOrganizer = call organizerPk
+    * def getAttendee = call attendeePk
+
+    * def laoValidCreationTime = function(){ return 1633035721 }
```
Comment (GabrielFleischer) [inline tests/karate/src/test/java/be/constants.feature:19]: What does this time corresponds to ?
Beware that in the future, backends might add checks on the validity of this value. (Not too far in the past nor in the future)

Maybe add a small comment explaining that.

File: tests/karate/src/test/java/be/constants.feature [hunk 4]
```diff
@@ -2,34 +2,55 @@
...
+    * def getRolCallValidName = call rollCallValidName
+
+    * def getLaoValidCreationTime = call laoValidCreationTime
+    * def getLaoInvalidCreationTime = call laoInvalidCreationTime
+    * def getRollCallValidCreationTime = call rollCallValidCreationTime
+
+    * def constructLaoId =
+       """
+          function(laoName, time){
+            var JsonConverter = Java.type('be.utils.JsonConverter')
```
Comment (GabrielFleischer) [inline tests/karate/src/test/java/be/constants.feature:38]: `JsonConverter` should definitely be renamed 😆

File: tests/karate/src/test/java/be/constants.feature [hunk 5]
```diff
@@ -116,31 +135,19 @@ Feature: Constants
...
+    * def getRollCallInvalidId = constructRollCallId(getLaoValid, getLaoValidName, 1633098853)
 
     * def getRollCallOpenValidId = call createValidRollCallOpenId
-    * def getRollCallOpenValidUpdateId = call createValidRollCallOpenUpdateId
+    * def getRollCallOpenValidUpdateId = constructRollCallUpdateId(getLaoValid, getRollCallOpenValidId, 1633099127)
     * def getRollCallOpenInvalidUpdateId = call createInvalidRollCallOpenUpdateId
 
     * def getRollCallCloseValidId = call createValidRollCallCloseId
-    * def getRollCallCloseValidUpdateId = call createValidRollCallCloseUpdateId
+    * def getRollCallCloseValidUpdateId = constructRollCallUpdateId(getLaoValid, getRollCallCloseValidId, 1633099135)
```
Comment (GabrielFleischer) [inline tests/karate/src/test/java/be/constants.feature:150]: Why are you using hard-coded values here ? What is their meaning ?

File: tests/karate/src/test/java/be/rollCall/createRollCall.feature [hunk 6]
```diff
@@ -46,6 +45,7 @@ Feature: Create a Roll Call
   # an empty roll call name should result in an error message from the backend.
   Scenario: Roll Call Creation with empty name should return an error code
     Given call read('classpath:be/utils/simpleScenarios.feature@name=valid_lao')
+    * def rollCallEmptyName = ""
```
Comment (GabrielFleischer) [inline tests/karate/src/test/java/be/rollCall/createRollCall.feature:48]: Is this variable used ?
I cannot find it.

File: tests/karate/src/test/java/be/rollCall/createRollCall.feature [hunk 7]
```diff
@@ -142,14 +142,15 @@ Feature: Create a Roll Call
...
     And def validCreateRollCall =
       """
         {
           "object": "roll_call",
           "action": "create",
-          "id": '#(getRollCallValidId)',
+          "id": '#(getRollCallValidId(rollCallName, creationTimeNegative))',
           "name": "Roll Call ",
-          "creation":-153,
+          "creation": -153,
```
Comment (GabrielFleischer) [inline tests/karate/src/test/java/be/rollCall/createRollCall.feature:153]: You should reuse the variable you just created

File: tests/karate/src/test/java/be/rollCall/openRollCall.feature [hunk 8]
```diff
@@ -36,6 +36,9 @@ Feature: Roll Call Open
           "opened_at": 1633099127
         }
       """
+    * karate.log("------------4444444444444444------------------------")
+    * karate.log(validOpenRollCall)
+    * karate.log("------------4444444444444444------------------------")
```
Comment (GabrielFleischer) [inline tests/karate/src/test/java/be/rollCall/openRollCall.feature:41]: ?

**Review by GabrielFleischer — APPROVED** [review]
Great work !

Jut some small issue and typos here and there.

**Review by GabrielFleischer — CHANGES_REQUESTED** [review]
Woops, it was meant to request changes

File: tests/karate/src/test/java/be/rollCall/createRollCall.feature (hunk 6 above)
Comment (Tesa27) [reply tests/karate/src/test/java/be/rollCall/createRollCall.feature:48]: Forgot to erase is, thank you for noticing

**Review by GabrielFleischer — APPROVED** [review]
LGTM

**Pull request merged**